zcncore/transaction_query_base.go

```diff
@@ -184,6 +188,117 @@ func (tq *TransactionQuery) randOne(ctx context.Context) (string, error) {
 	}
 }
 
+func (tq *TransactionQuery) FromConsensus(ctx context.Context, query string, handle QueryResultHandle) error {
+
+	min := getMinShardersVerify()
+
+	if err := tq.validate(min); err != nil {
+		return err
+	}
+
+	next := make(chan string, 1)
+	result := make(chan QueryResult, min)
+	done := make(chan bool, 1)
+	defer func() {
+		// cancel next channel
+		done <- true
+	}()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-done:
+				return
+			case sharder := <-next:
+				go func(sharder string) {
+
+					err := tq.checkHealth(ctx, sharder)
```

Comment:
No need to do health check as it will double requests number and give pressure to the servers.

Reply:
removed